codex/sales/states/downloading.nim

```diff
@@ -38,5 +46,49 @@ method run*(state: SaleDownloading, machine: Machine): Future[?State] {.async.}
   without request =? data.request:
     raiseAssert "no sale request"
 
-  await onStore(request, data.slotIndex, data.availability)
+  without availability =? await reservations.find(
+      request.ask.slotSize,
+      request.ask.duration,
+      request.ask.pricePerSlot,
+      used = false):
+    info "no availability found for request, ignoring",
+      slotSize = request.ask.slotSize,
+      duration = request.ask.duration,
+      pricePerSlot = request.ask.pricePerSlot,
+      used = false
+    return some State(SaleFinished())
```

Comment:
When there is no availability, we shouldn't move to the "finished" state. The finished state signals that the sale was successful, and that we should start providing storage proofs.

Reply:
True, we should perhaps define a start where eg `SaleIgnored` such that the state of an agent doesn't remain on `SaleDownloading` when it has been ignored.

Reply:
Perhaps we'd also like to remove the sales agent from the list of agents in the sales module once Finished/Ignored?